### PR TITLE
precisionFromString update

### PR DESCRIPTION
**Pull request by ttodua**
base function `precisionFromString` works with such numbers `'0.001'` but not with `'1e-3'` . 
so, this update is needed

### CODE REVIEW

**Comment by ttodua** [comment]
all methods / langs are ready now to be merged.

**Comment by carlosmiei** [comment]
@ttodua the edits in this PR broke the build, can you take a look?
image

**Comment by ttodua** [comment]
> @ttodua the edits in this PR broke the build, can you take a look? image

hm, the reason turns out that digifinex provides precision strings like `1.1e-05` which were not handled in py base well. creating a fix.

File: js/base/functions/number.js [hunk 1]
```diff
@@ -87,8 +87,18 @@ const truncate_regExpCache = []
...
+    if (str.indexOf ('e') > -1) {
+        const numStr = str.replace (/\de/, '')
+        return parseInt (numStr) * -1
+    }
+    // support integer formats (without dot) like '1', '10' etc
+    if (str.indexOf ('.') === -1) {
+        return str.length * -1
+    }
+    // default strings like '0.0001'
+    const split = str.replace (/0+$/g, '').split ('.')
```
Comment (ttodua) [inline js/base/functions/number.js:101]: this 101th and next line was not touched.

**Comment by ttodua** [comment]
@carlosmiei fixed. please review now.

**Comment by carlosmiei** [comment]
LGTM ✅

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**